ios/CodePush/CodePushUpdateUtils.m

```diff
@@ -20,6 +20,14 @@ + (void)addContentsOfFolderToManifest:(NSString *)folderPath
     }
     
     for (NSString *fileName in folderFiles) {
+#if TARGET_IPHONE_SIMULATOR
```

Comment:
So I believe this error will also happen if the package you release was created on OS X (with accompanying `.DS_Store`), but downloaded on a real mobile device, because the server's hashing algorithm explicitly ignores `.DS_Store` and any `__MACOSX/` directory at the root.

In other words, the problem here is an inconsistency between the hashing algorithm used on the server and the hashing algorithm used on the client. So, I think you can remove this `#ifdef`, and if you like, you can also check for the `__MACOSX` directory, but that's up to you (I'm happy to fix it afterwards otherwise).